webapp/src/components/App/Singlestudy/explore/Configuration/General/dialogs/ScenarioPlaylistDialog/YearsSelectionFE/utils.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2025, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+import { isNumericValue } from "@/utils/numberUtils";
+import * as R from "ramda";
+import * as RA from "ramda-adjunct";
+
+////////////////////////////////////////////////////////////////
+// Types
+////////////////////////////////////////////////////////////////
+
+export type SelectionType = "all" | "range" | "advanced";
+
+export type Range = [number, number];
+export type Selection = number | Range;
+
+////////////////////////////////////////////////////////////////
+// Constants
+////////////////////////////////////////////////////////////////
+
+export const SELECTIONS_SEPARATOR = "," as const;
+export const RANGE_SEPARATOR = "-" as const;
+
+////////////////////////////////////////////////////////////////
+// Functions
+////////////////////////////////////////////////////////////////
+
+export function selectionsToNumbers(selections: Selection[]) {
+  const numbersSet = new Set<number>();
+
+  selections.forEach((selection) => {
+    if (Array.isArray(selection)) {
+      const [start, end] = selection;
+      R.range(start, end + 1).map((num) => numbersSet.add(num));
+    } else {
+      numbersSet.add(selection);
+    }
+  });
+
+  return Array.from(numbersSet).sort((a, b) => a - b);
+}
```

Comment:
optional optimisation: you can use `for..of` (faster for simple iterations + the return value of the `map()` operation is not being captured or used) and reduce nesting
```ts
export function selectionsToNumbers(selections: Selection[]) {
  const numbersSet = new Set<number>();

  for (const selection of selections) {
    if (!Array.isArray(selection)) {
      numbersSet.add(selection);
      continue;
    }
    
    const [start, end] = selection;
    const range = R.range(start, end + 1);
    
    for (const num of range) {
      numbersSet.add(num);
    }
  }

  return Array.from(numbersSet).sort((a, b) => a - b);
}
```

Reply:
I changed the `map` with `forEach`